imx/mint.go

```diff
@@ -95,9 +102,38 @@ func (c *Client) Mint(
 		}
 	}
 
-	mintTokensResponse, httpResp, err := c.MintTokens(ctx).MintTokensRequestV2(mintRequest).Execute()
+	mintTokensResponse, httpResponse, err := c.mintsAPI.MintTokens(ctx).MintTokensRequestV2(mintRequest).Execute()
 	if err != nil {
-		return nil, fmt.Errorf("error when calling `api.Mints.MintTokens`: %v, HTTP response body: %v", err, httpResp.Body)
+		return nil, NewAPIError(httpResponse, err)
 	}
 	return mintTokensResponse, nil
 }
+
+/*
+GetMint Get details of a mint with the given ID
```

Comment:
I don't think you need to include the function name as a prefix to these titles, for any of these annotations. (I've also noticed some inconsistencies)

e.g this is fine:

```golang
/*
Get details of a Mint with the given ID
@param ctx context.Context - for cancellation, deadlines, tracing, etc or context.Background().
@param id Mint ID. This is the transaction_id returned from listMints
@return ApiGetMintRequest
*/
```

Reply:
We do, otherwise go linter complains.

Reply:
fair enough. we don't really have a "source of truth" as of yet, but could we also use [this](https://github.com/immutable/imx-core-sdk/blob/coresdk-interface-class/src/ImmutableX.ts) as inspiration for better/more concise descriptions (these have been reviewed more thoroughly)

Reply:
I have added them as I see fit happy to be re-sued and also corrected if it does not make sense. I am not sure if it has been reviewed more throughly.

Reply:
For descriptions it'd make sense to be consistent across SDKs. We can treat the TS SDK as SoT. 

Can be done separately once the TS SDK 1.0 is released.

Reply:
@matthewmuscat sorry did not notice the link to this until now.